### ISSUE TITLE
Assorted offline cache issues

### ISSUE THREAD

**Issue by jywarren** [publiclab/image-sequencer#605]
Just reporting a few issues with the offline cache function, maybe related to #544. 

We can break out solutions and sub-issues from here. Just want to say the offline cache is AMAZING and has saved me a number of times when demonstrating this to people. Just looking to expand and refine it. Thank you!!!

* [ ] as in below image, i see 2 links to clear cache
* [ ] the version numbering hasn't changed in a while. Do we have to manually update this in `examples/sw.js`? I tried but to no effect
* [ ] neither button seems to update the cache
* [ ] maybe we need it to say something a little more specific, like, `A new version is available. Click to update.`
* [ ] we should document how this works somewhere so we can maintain it
* [ ] could we tie the cache version to the npm version?
* [ ] can we isolate all the cache-related code as much as possible into `/examples/lib/cache/...`?

![image](https://user-images.githubusercontent.com/24359/50707921-907b8700-1030-11e9-9b54-e4d1dfec6f93.png)

**Comment by Mridul97** [comment]
@jywarren the cache seems to be updating, please see the below gif,

![cache](https://user-images.githubusercontent.com/21158457/50722849-7e1a4f80-10fb-11e9-9a02-c5fe370cb75c.gif)

To change the cache version number we have to change it manually in sw.js file and then it will get update when we go to the website next time, it does not updates on reload. Please see the gif below : 

![cache2](https://user-images.githubusercontent.com/21158457/50722940-c128f280-10fc-11e9-8c9a-85dbf0622146.gif)

We can tie the npm version to the cache version, if we can get the value of npm version in sw.js. Do we manually update the value of npm version?
 
I like the idea to isolate the cache code and would love to work on it! I will remove the other cache button, as it has no use. I would like to document the working as well. Thanks!

**Comment by VibhorCodecianGupta** [comment]
Yes, that's exactly what I noticed too, the cache works fine for me.

**Comment by HarshKhandeparkar** [comment]
Some issues are fixed and docs part is mentioned in another issue.

**Comment by jywarren** [comment]
Hi all, regarding cache, we don't have an easy way to tell if it's being used, when it was updated, or what version we're using in the demo. I think we should consider turning caching off until we can address these as they may be preventing us from diagnosing and detecting these kinds of issues:

https://github.com/publiclab/image-sequencer/issues/1061

**Comment by jywarren** [comment]
I've updated the issue above to highlight our main problems and added a few more steps that could be broken out.

**Comment by HarshKhandeparkar** [comment]
@jywarren I think we can do something in the long term to improve the service worker, we can build it dynamically using maybe browserify or something. We can keep the code separately and maybe make a grunt task to compile it.

**Comment by HarshKhandeparkar** [comment]
Some parts were completed in #1364

**Comment by jywarren** [comment]
Updated to show 2 remaining pieces - the checksum of the last commit and the date of last update. Thanks!

**Comment by jywarren** [comment]
Confirming this worked (except 2 remaining pieces) for v3.7.1:

![Screen Shot 2022-01-15 at 2 25 29 PM](https://user-images.githubusercontent.com/24359/149635361-3c6653c5-e953-4207-a5bc-4d27cc53c7ba.png)
![Screen Shot 2022-01-15 at 2 25 39 PM](https://user-images.githubusercontent.com/24359/149635364-0bb9aa0e-f6e6-4905-a16c-602e7d188f28.png)

![Screen Shot 2022-01-15 at 2 25 42 PM](https://user-images.githubusercontent.com/24359/149635367-db51e523-0595-443e-9ecc-744493991d08.png)
